sherpa/fit.py

```diff
@@ -1364,19 +1392,21 @@ def fit(self,
         Repeat the fit, after resetting the model, so we can see how
         the optimiser searched the parameter space:
 
-        >>> from io import StringIO
         >>> m.reset()
-        >>> optdata = StringIO()
-        >>> out2 = f.fit(outfile=optdata)
-        >>> print(optdata.getvalue())
-        # nfev statistic const1d.c0
-        0.000000e+00 1.900000e+02 1.000000e+00
-        1.000000e+00 1.900000e+02 1.000000e+00
-        2.000000e+00 1.899834e+02 1.000345e+00
-        3.000000e+00 4.600000e+01 7.000000e+00
-        4.000000e+00 4.600002e+01 7.002417e+00
-        5.000000e+00 4.600000e+01 7.000000e+00
-
+        >>> out = f.fit(record_steps=True)
+        >>> for n, stat, c0 in zip(out.record_steps['nfev'],
+        ...                          out.record_steps['statistic'],
+        ...                          out.record_steps['const1d.c0']):
+        ...     print(f"{n}, {stat:6.6e}, {c0:8.6f}")
```

Comment:
The f-string has commas after the columns but the output below doesn't have them. Shouldn't the doc-testing have caught this?

Reply:
It would have caught that, if we run doctests on it but `sherpa/fit.py` is still on the balcklist for for doctests (see list in `purest.ini` because we never made all the other docstrings in the file pass - but this is exactly why I want to increase coverage of docstrings!